packages/core/src/enqueue_links/enqueue_links.ts

```diff
@@ -212,8 +212,10 @@ export async function enqueueLinks(options: EnqueueLinksOptions): Promise<BatchA
         urlPatternObjects.push(...constructRegExpObjectsFromRegExps(regexps));
     }
 
+    options.strategy ??= urlPatternObjects.length > 0 ? undefined : EnqueueStrategy.SameSubdomain;
```

Comment:
Maybe wrap in an if check instead? 
```suggestion
    if (urlPatternObjects.length > 0) {
    	options.strategy ??= EnqueueStrategy.SameSubdomain;
	}
```

Reply:
you can improve on this as part of #64, i've done this only so the example e2e test works correctly

(btw my code does the opposite than your suggestion, we don't want the default to be set if there are any url patterns)